www/js/components/pages/vm/disks.js

```diff
@@ -10,43 +10,133 @@
 
+/**
+ * Disks Table/List
+ */
+var Disks = Backbone.Collection.extend({});
+var JobProgressView = require('../../../views/job-progress');
+var DisksFormView = require('./disks-form');
 var DiskRowTemplate = require('./disks-row.hbs');
+
+var adminui = require('../../../adminui');
 var DiskRow = Backbone.Marionette.ItemView.extend({
     tagName: 'tr',
     template: DiskRowTemplate,
     events: {
+        'click .edit': 'resizeDisk',
+        'click .delete': 'deleteDisk'
     },
-    initialize: function (options)  {
-        // Override list initialization
+
+    deleteDisk: function () {
+        if (window.confirm('Are you sure you want to delete this disk?')) {
+            this.vm.removeDisk(this.model.get('pci_slot'),
+                this._onJob.bind(this));
+        }
+    },
+    _onJob: function (err, job) {
+        if (err) {
+            console.log('[job error]: ', err);
+            var msg = (err.responseText) ?
+                    JSON.parse(err.responseText).message:
+                    'Error creating job';
+            window.alert(msg);
+            return;
+        }
+        var self = this;
+        var jobView = new JobProgressView({model: job});
+        this.listenTo(jobView, 'execution', function (exec) {
+            if (exec === 'succeeded') {
+                self.vm.fetch();
+            }
+        });
+        jobView.show();
+    },
+
+    resizeDisk: function () {
+        var self = this;
+        var view = new DisksFormView({
+            vm: self.vm,
+            model: self.model
+        });
+        view.render();
     }
+
 });
 
 var DisksView = Backbone.Marionette.CompositeView.extend({
-    template: require('./disks.hbs'),
     itemView: DiskRow,
     itemViewContainer: 'tbody',
+    template: require('./disks.hbs'),
     attributes: {
         id: 'vm-disks'
     },
     events: {
+        'click button': 'onClickAddDisk'
     },
 
     initialize: function (options)  {
         this.vm = options.vm;
         this.collection = new Disks(this.vm.get('disks'));
+        this.listenTo(this.collection, "add", this.render, this);
+        this.listenTo(this.collection, "remove", this.render, this);
+        this.listenTo(this.collection, "sync", this.render, this);
+        this.listenTo(this.vm, 'change:disks', this.resetDisks, this);
+    },
+
+    resetDisks: function(vm, n) {
+        this.collection.reset(vm.get('disks'));
+        this.render();
+    },
+
+    onBeforeItemAdded: function(view) {
+        view.vm = this.vm;
     },
 
     templateHelpers: function() {
         var self = this;
         return {
             disks: function() {
                 return self.collection;
+            },
+
+            hasDisks: function(data) {
+                return self.collection.length > 0;
             }
         };
-    }
+    },
+
+    onClickAddDisk: function () {
+        var self = this;
+        var view = new DisksFormView({
+            vm: self.vm
```

Comment:
Do you need a `model` attribute here? As on line 60 it uses:
```
vm: self.vm,
model: self.model
```

Reply:
No, b/c on this case the disk is new and we just need the `size` parameter from the form. So, we can just build a model with that single attribute.